### PR TITLE
Split skb if can't allocate a new response

### PR DESCRIPTION
**Pull request by ikoveshnikov**
If sibling response can't be allocated or paired with any request, then splitting skb is required to send client exactly one response, but not response and rest of data in skb.

### CODE REVIEW

File: tempesta_fw/http.c [hunk 1]
```diff
@@ -1818,14 +1818,21 @@ tfw_http_msg_create_sibling(TfwHttpMsg *hm, struct sk_buff **skb,
...
 	/* The sibling message belongs to the same connection. */
 	shm = (TfwHttpMsg *)tfw_http_conn_msg_alloc(hm->conn);
-	if (unlikely(!shm))
+	if (unlikely(!shm)) {
+		/*
+		 * Split skb to proceed with current message @hm, or rest of
+		 * data in this connection will be attached to the message.
+		 */
+		if (type == Conn_Srv)
+			*skb = ss_skb_split(*skb, split_offset);
```
Comment (krizhanovsky) [inline tempesta_fw/http.c]: Yeah, makes sense. But what about 2 pipelined requests? It seems we send then as one request to the server and the server returns us 2 responses - we can't find a request for the 2nd response and get #941...

File: tempesta_fw/http.c [hunk 2]
```diff
@@ -1818,14 +1818,21 @@ tfw_http_msg_create_sibling(TfwHttpMsg *hm, struct sk_buff **skb,
...
 		return NULL;
+	}
 
 	/*
 	 * New message created, so it should be in whitelist if
 	 * previous message was (for client connections).
 	 */
-	if (TFW_CONN_TYPE(hm->conn) & Conn_Clnt)
+	if (type == Conn_Clnt)
```
Comment (krizhanovsky) [inline tempesta_fw/http.c]: `type` was unused in the function and I think it's better to remove the argument: `hm->conn->proto.type` is quick after all.

File: tempesta_fw/http.c (hunk 1 above)
Comment (ikoveshnikov) [reply tempesta_fw/http.c]: If we can't allocate a sibling message, it means that we have to drop the connection. For server connection we can use message already parsed from the connection. It's normal response and we can forward it to the client. Just need to set correct framing. And the patch is about it.

Situation is little different with client connections. As I said the connection is closed, so we drop the already processed request without further forwarding to backend server. There is no sense in splitting skb for client connections.
https://github.com/tempesta-tech/tempesta/blob/4935217018d37315f24166e38d936a0fbeb3d6bc/tempesta_fw/http.c#L2718-L2732

That's not a reason for #941

Comment (krizhanovsky) [reply tempesta_fw/http.c]: Ok, has sense. Please add a short comment that we do not care about requests there and just drop client connections

Comment (krizhanovsky) [reply tempesta_fw/http.c]: Ok, has sense. Please add a short comment that we do not care about requests there and just drop client connections

**Pull request merged**